main.go

```diff
@@ -42,15 +43,69 @@ func main() {
 	tc := oauth2.NewClient(ctx, ts)
 	client := github.NewClient(tc)
 
-	// Get the Pull Request details
+	// Fetch PR details
 	pr, _, err := client.PullRequests.Get(ctx, *owner, *repo, *prNumber)
 	if err != nil {
 		fmt.Printf("Error fetching PR details: %v\n", err)
 		os.Exit(1)
 	}
 
-	// Generate a review using GPT-4
-	review, err := generateReviewWithAssistant(pr)
+	// Fetch the current user (the reviewer)
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		fmt.Printf("Error fetching user details: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Fetch PR checks (e.g., CI tests)
+	checks, _, err := client.Checks.ListCheckRunsForRef(ctx, *owner, *repo, *pr.Head.SHA, &github.ListCheckRunsOptions{})
+	if err != nil {
+		fmt.Printf("Error fetching PR checks: %v\n", err)
+		os.Exit(1)
+	}
+
+	// If any check has failed, do not allow approval
+	checksPassed := true
+	for _, check := range checks.CheckRuns {
+		if check.GetConclusion() == "failure" {
+			checksPassed = false
+			break
+		}
+	}
+
+	// Fetch PR files
+	files, _, err := client.PullRequests.ListFiles(ctx, *owner, *repo, *prNumber, &github.ListOptions{})
+	if err != nil {
+		fmt.Printf("Error fetching PR files: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Check for pending reviews
+	pendingReview, err := getPendingReview(client, ctx, *owner, *repo, *prNumber)
+	if err != nil {
+		fmt.Printf("Error checking for pending reviews: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Handle existing pending review
+	if pendingReview != nil {
+		fmt.Println("A pending review already exists.")
+		if *dryRun {
+			fmt.Println("Dry run: Review not posted to GitHub.")
+			return
+		}
+
+		// Optionally, submit or dismiss the pending review here
+		// For now, we'll dismiss it to proceed with the new review
+		err = dismissPendingReview(client, ctx, *owner, *repo, *prNumber, pendingReview.GetID(), "Dismissing pending review to submit a new one.")
+		if err != nil {
+			fmt.Printf("Error dismissing pending review: %v\n", err)
+			os.Exit(1)
+		}
```

Comment:
"The message 'A pending review already exists.' does not specify how to resolve it (i.e., by dismissing or completing the existing review). An actionable suggestion would improve UX.